src/App.js

```diff
@@ -80,13 +80,14 @@ const removeSong = (id) =>{
     let autoEncoded = encodeURIComponent(text)
     let jsonResults = await (searchSongs(autoEncoded));
     console.log('RESULTS: '+JSON.stringify(jsonResults));
-    for(let i=0;i<jsonResults.length;i++){
-      console.log('key: '+ jsonResults[i].id);
-      setTrackList((prev) =>
-         [...prev, <Track key = {jsonResults[i].id} accessKey = {jsonResults[i].id} song = {jsonResults[i].name} artist = {jsonResults[i].artist} album = {jsonResults[i].album} albumCover = {jsonResults[i].albumCover} addSong = {addSong} setPlaylistTracksData = {setPlaylistTracksData} isAdd={true}/>]
-      );
+    // for(let i=0;i<jsonResults.length;i++){
+    //   console.log('key: '+ jsonResults[i].id);
+    //   setTrackList((prev) =>
+    //      [...prev, <Track key = {jsonResults[i].id} accessKey = {jsonResults[i].id} song = {jsonResults[i].name} artist = {jsonResults[i].artist} album = {jsonResults[i].album} albumCover = {jsonResults[i].albumCover} addSong = {addSong} setPlaylistTracksData = {setPlaylistTracksData} isAdd={true}/>]
+    //   );
       
-    }
+    // }
+    setTrackList(jsonResults);
```

Comment:
Someone mentioned it in the StackOverflow thread, but typically you don't want to store JSX in your state. Here I'm just saving the array of track objects into our state for us to map over later.